src/main/java/animals/controller/Game.java

```diff
@@ -0,0 +1,162 @@
+                case KNOWLEDGE_TREE:
+                    kts.printTree(fs.load(type, FILE_NAME));
+                    System.out.println();
+                    break;
+                case EXIT:
+                    quit = true;
+                    SCANNER.close();
+                    System.out.println(getLocalString("bye"));
+                    break;
+                case UNDEFINED:
+                    System.out.println(getLocalString("wrong"));
+            }
+        }
+    }
+
+    public void play(Node node) {
+        System.out.println(node.askQ());
+        if (!node.isLeaf()) {
+            if (askYesNo()) {
+                play(node.getYes());
+            } else {
+                play(node.getNo());
+            }
+        } else {
+            if (!askYesNo()) {
+                String firstAnimal = node.getData();
+                System.out.println(getLocalString("give.up"));
+                String userNewAnimal = lr.getAnimal(SCANNER.nextLine().strip().toLowerCase());
+
+                String positive = lr.getStatement(firstAnimal, userNewAnimal);
+                String negative = lr.toNegative(positive);
+                System.out.println(getLocalString("is.correct") + userNewAnimal + "?");
+
+                String learn1 = lr.toAnimalFact(positive, firstAnimal);
+                String learn1Neg = lr.toAnimalFact(negative, firstAnimal);
+                String learn2 = lr.toAnimalFact(positive, userNewAnimal);
+                String learn2Neg = lr.toAnimalFact(negative, userNewAnimal);
+                String question = lr.getQuestion(positive);
+                Node secondNode = new Node(userNewAnimal);
+                if (!askYesNo()) {
+                    node.yes = new Node(node.getData());
+                    node.data = question;
+                    node.no = secondNode;
+                    node.setLeaf(false);
+                    System.out.println(lr.getLearnedPhrase(learn1, learn2Neg, question));
+                } else {
+                    node.no = new Node(node.data);
+                    node.data = question;
+                    node.yes = secondNode;
+                    node.setLeaf(false);
+                    System.out.println(lr.getLearnedPhrase(learn1Neg, learn2, question));
+                }
+            }
+        }
+    }
+
+    public boolean askYesNo() {
+        final var bundle = ResourceBundle.getBundle("application");
+        while (true) {
+            String answer = SCANNER.nextLine().trim();
+            if (answer.matches(bundle.getString("pattern.yes"))) {
+                return true;
+            }
+            if (answer.matches(bundle.getString("pattern.no"))) {
+                return false;
+            }
+            System.out.println(bundle.getString("ask.again"));
+        }
+    }
+
+    public String checkTimeOfDay() {
+        LocalTime now = LocalTime.now();
+        if (now.isAfter(MORNING_START) && now.isBefore(MORNING_END)) {
```

Comment:
“Good morning” should be used from 5:00 am to 12:00 pm, “Good afternoon” is best for the period between 12:00 pm and 6:00 pm, and “Good evening” is used after 6 p.m. 
Please check all edge cases: 5:00 am, 12:00 pm, 6:00 pm

Reply:
It still doesn't work correctly for 12:00
Please run locally for `LocalTime now = LocalTime.of(12, 0);` and check why it happens.